smart_contracts/contracts/TradeDxt.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity ^0.8.0;
+import "./Dextr.sol";
+import "./TradeHelper.sol";
+
+contract TradeDxt is Dextr, TradeHelper {
+    modifier onlyOwner() override(Dextr, TradeHelper) {
+        require(msg.sender == owner, "Not owner of contract, cannot call function.");
+        _;
+    }
+    constructor() payable {
+        owner = msg.sender;
+        //payable(address(this)).send(1000000000000 wei);
+        //_mint(address(this), 1000000 * (10 ** uint256(decimals())));
+    }
+    modifier nonZeroAddress(address _receiver) {
+        require(_receiver!=address(0), "Non-zero address check");
+        _;
+    }
+    function buyDxt(address _buyer, uint256 _amount) external payable {
+        //call from buyer's address
+        require(msg.sender == _buyer && _amount < totalSupply(), "Error: Not enough Ether in address"); // && _convertToEth(_amount) > msg.value
```

Comment:
why require(msg.sender == _buyer) ?

If msg.sender is always going to be the buyer then why even ask the user to pass their own address?